scrapy/http/response/text.py

```diff
@@ -158,6 +159,66 @@ def follow(self, url, callback=None, method='GET', headers=None, body=None,
             cb_kwargs=cb_kwargs,
         )
 
+    def follow_all(self, urls=None, callback=None, method='GET', headers=None, body=None,
+                   cookies=None, meta=None, encoding=None, priority=0,
+                   dont_filter=False, errback=None, cb_kwargs=None,
+                   css=None, xpath=None):
+        # type: (...) -> Generator[Request, None, None]
+        """
+        A generator that produces :class:`~.Request` instances to follow all
+        links in ``urls``. It accepts the same arguments as the :class:`~.Request`
+        initializer, except that each ``urls`` element does not need to be an absolute
```

Comment:
My understanding is that `__new__` are constructors, and `__init__` are initializers, although this doesn't seem to matter much in practice in most of the docs I have read. Should we sacrifice correctness in favour of clarity?

Reply:
I’ve not been able to find a reference to ‘initializer’ in https://docs.python.org/3/reference/datamodel.html, but looking at https://docs.python.org/3/reference/datamodel.html#object.__new__ ‘constructor’ is not appropriate either, as it seems to be used to refer to the class itself when used to create an instance.

Maybe we should explicitly say ‘`__init__` method’.

Reply:
I suspect the official docs might intentionally skip the "constructor" and "initializer" names, to avoid this kind of confusion (and perhaps because even the core devs do not agree on them 😅)
You are correct, mentioning `__init__` sounds good, thanks!